### PR TITLE
Add some tests to K8s deploy watcher.

### PR DESCRIPTION
**Pull request by henders**
Just adding some tests to the K8s plugin deploy watcher class.

/cc @sbrnunes @msufa 

### References
 - Jira link: 

### Risks
 - None

### CODE REVIEW

File: plugins/kubernetes/test/models/watchers/deploy_watcher_test.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  describe '#watch' do
+    it 'sets deploy as finished when all pods active' do
+      watcher = Watchers::DeployWatcher.new(deploying_release)
+      deploying_release.release_docs.each do |release_doc|
+        release_doc.replica_target.times do |i|
+          msg = create_msg(name: "foo-#{i}", ready: 'True')
+          watcher.handle_update(release_doc.replication_controller_name, msg)
+        end
+      end
+      watcher.send(:deploy_finished?).must_equal true
```
Comment (msufa) [inline plugins/kubernetes/test/models/watchers/deploy_watcher_test.rb]: You can just say `watcher.deploy_finished?`

Comment (henders) [inline plugins/kubernetes/test/models/watchers/deploy_watcher_test.rb]: nope, it's a private method.

Comment (msufa) [inline plugins/kubernetes/test/models/watchers/deploy_watcher_test.rb]: D'oh, I thought you were trying to do some async magic.

**Comment by msufa** [comment]
:+1: on :green_heart:

**Comment by henders** [comment]
Yup, that :green_heart: will be harder to come by, will need to change Samson over to use DatabaseCleaner to get rid of the default transactions.  Celluloid tests are failing due to DB locks while the main test transaction is open.

**Comment by henders** [comment]
@msufa , @sbrnunes , @jonmoter 
Added some dodginess to the tests to allow threaded tests to succeed with transactions. Another +1 desired!

**Comment by sbrnunes** [comment]
I haven't seen any problems from this change, so, I wouldn't mind to give you a :+1:. 
However, I would prefer if someone with more experience than me could take a look at it as well: @jonmoter, @steved ?

This PR is required in order to get #717 and #700 merged.

**Comment by sbrnunes** [comment]
Also, @henders, regarding the conflicts, at this stage, I think you should be able to just revert those tests and let the ones in master prevail.

**Comment by steved** [comment]
I know nothing about this code, but I'm not a huge fan of such a wide-reaching patch. The only solutions I can think of are DI and more fine-grained unit tests with stubbing for the threaded updates.

**Pull request closed without merging**